### PR TITLE
Setup Accelerate and MKL for 32-bit, MKL getrf, and Metal.jl integration

### PR DESCRIPTION
**Pull request by ChrisRackauckas**
Kind of a lot in a single PR but this is what I ended up with after a flight

### CODE REVIEW

**Comment by christiangnrd** [comment]
Tried to run the Metal benchmark and I'm getting this error:

`ERROR: MethodError: no method matching do_factorization(::MetalLUFactorization, ::Matrix{Float32}, ::Vector{Float32}, ::Vector{Float32})`

**Comment by ChrisRackauckas** [comment]
You need to re-instantiate for it to bring in the extension from a PR branch.

File: benchmarks/cudalu.jl [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+                     sum((k + 1):n) do j
+            isempty((k + 1):m) ? 0 : sum((k + 1):m) do i
+                innerflop
+            end
+        end
+        invflop + scaleflop + updateflop
+    end
+end
+
+algs = [MKLLUFactorization(), CUDAOffloadFactorization()]
```
Comment (christiangnrd) [inline benchmarks/cudalu.jl:21]:
```suggestion
algs = [MKLLUFactorization(), CudaOffloadFactorization()]
```

**Comment by ViralBShah** [comment]
Why do we need MKL 32-bit?

**Comment by ChrisRackauckas** [comment]
GPU comparison, Neural ODE training, mixed precision algorithms, etc.

**Pull request merged**